contracts/CumulativeMerkleDrop.sol

```diff
@@ -25,24 +25,22 @@ contract CumulativeMerkleDrop is Ownable, ICumulativeMerkleDrop {
         bytes32[] proof;
     }
 
-    event RewardsHolderUpdated(address oldRewardsHolder, address newRewardsHolder);
-
     constructor(address token_, address rewardsHolder_) {
         require(IERC20(token_).totalSupply() > 0, "Token contract must be set");
-        require(rewardsHolder_ != address(0), "Rewards GHolder must be an address");
+        require(rewardsHolder_ != address(0), "Rewards holder must be an address");
         token = token_;
         rewardsHolder = rewardsHolder_;
     }
 
     function setMerkleRoot(bytes32 merkleRoot_) external override onlyOwner {
-        emit MerkelRootUpdated(merkleRoot, merkleRoot_);
         merkleRoot = merkleRoot_;
+        emit MerkelRootUpdated(merkleRoot, merkleRoot_);
```

Comment:
this isn't quite correct, because now `merkleRoot` and `merkleRoot_` are the same

Reply:
It's true!